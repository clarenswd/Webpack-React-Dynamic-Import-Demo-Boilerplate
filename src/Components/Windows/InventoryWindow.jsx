import React from 'react';
import PropTypes from 'prop-types';
import style from './InventoryWindow.scss?module';

const InventoryWindow = (props) => {
  const {
    children,
    Shell,
  } = props;

  // console.log(children.length)

  return (
    <Shell className={style.container}>
      {children}
    </Shell>
  );
};

InventoryWindow.propTypes = {
  children: PropTypes.node.isRequired,
  Shell: PropTypes.node,
};

InventoryWindow.defaultProps = {
  Shell: 'div',
};

export default InventoryWindow;
