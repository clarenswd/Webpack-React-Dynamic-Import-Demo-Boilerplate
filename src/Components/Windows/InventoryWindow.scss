.container {
  border: 5px solid black;
  display: inline-grid;
  background-color: #000e57;
  grid-template-columns: repeat(2, 120px [inventory-col]);
  grid-template-rows: repeat(4, 60px [inventory-row]);
  grid-gap: 5px;
  padding: 5px;
}