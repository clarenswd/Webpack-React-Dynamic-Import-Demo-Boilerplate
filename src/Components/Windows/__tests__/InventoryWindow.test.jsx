import React from 'react';
import renderer from 'react-test-renderer';
import InventoryWindow from '../InventoryWindow';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <InventoryWindow>
        <div>Child 1</div>
        <div>Child 2</div>
      </InventoryWindow>
     )
    .toJSON();
  expect(tree).toMatchSnapshot();
});