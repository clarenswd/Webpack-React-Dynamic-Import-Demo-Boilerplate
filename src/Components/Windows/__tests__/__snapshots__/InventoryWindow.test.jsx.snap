// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div>
    Child 1
  </div>
  <div>
    Child 2
  </div>
</div>
`;
